pkg/values/map.go

```diff
@@ -69,6 +70,33 @@ func (m *Map) UnwrapTo(to any) error {
 	return d.Decode(m.Underlying)
 }
 
+// DeleteAtPath deletes a value from a map at a given dot separated path.  Returns true if an element at the given
+// path was found and deleted, false otherwise.
+func (m *Map) DeleteAtPath(path string) bool {
```

Comment:
this is only used in a test AFAICT. what's going on?

Reply:
its used in core, seemed like the best place for the logic to sit